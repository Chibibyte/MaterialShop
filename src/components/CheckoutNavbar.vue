<template>
  <v-app-bar app class="accent">
    <v-container app class="d-flex justify-space-around">
      <NavButton
        :disabled="$store.state.checkout.checkoutStates.summary"
        to="/"
        :icon="$constants.BUTTONICONS.CANCEL"
        name="Cancel"
      />

      <NavButton
        :disabled="!$store.state.checkout.checkoutStates.authentication"
        to="/checkout/address"
        :icon="$constants.BUTTONICONS.AUTHENTICATION"
        name="Delivery address"
      />
      <NavButton
        :disabled="!$store.state.checkout.checkoutStates.deliveryAddress"
        to="/checkout/bankaccount"
        :icon="$constants.BUTTONICONS.ADDRESS"
        name="Bank Account"
      />
      <NavButton
        :disabled="!$store.state.checkout.checkoutStates.bankaccount"
        to="/checkout/shipping"
        :icon="$constants.BUTTONICONS.BANKACCOUNT"
        name="Shipping"
      />
      <NavButton
        :disabled="!$store.state.checkout.checkoutStates.shipping"
        to="/checkout/payment"
        :icon="$constants.BUTTONICONS.SHIPPING"
        name="Payment"
      />
      <NavButton
        :disabled="!$store.state.checkout.checkoutStates.payment"
        to="/checkout/summary"
        :icon="$constants.BUTTONICONS.PAYMENT"
        name="Summary"
      />
    </v-container>
  </v-app-bar>
</template>

<script>
import NavButton from "@/components/NavButton.vue";
export default {
  name: "CheckoutNavbar",
  components: {
    NavButton,
  },
  methods: {
    cancelEnabled() {
      // let cancelEn = Object.values(
      //   this.$store.state.checkout.checkoutStates
      // ).reduce((acc, cur) => acc || cur);
      // return cancelEn;
      return true;
    },
  },
};
</script>