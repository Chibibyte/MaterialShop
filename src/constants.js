export default {
    BUTTONICONS: {
        AUTHENTICATION: 'mdi-login-variant',
        ADDRESS: 'mdi-package-variant',
        BANKACCOUNT: 'mdi-bank',
        SHIPPING: 'mdi-truck',
        PAYMENT: 'mdi-cash',
        SUMMARY: 'mdi-text-box-check-outline',
        CHECKOUT_COMPLETE: 'mdi-check',
        ADD: 'mdi-plus',
        REMOVE: 'mdi-minus',
        CANCEL: 'mdi-window-close',
        CONFIRM: 'mdi-check',
        SHOPPING: 'mdi-shopping',
        SUCCESS: 'mdi-check',
        FAILURE: 'mdi-window-close',
        WARNING: 'mdi-alert-circle-outline'
    }
}