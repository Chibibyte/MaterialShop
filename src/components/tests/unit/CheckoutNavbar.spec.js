
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CheckoutNavbar from '@/components/CheckoutNavbar.vue';
import constants from '@/constants.js';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.prototype.$constants = constants;

let store;
let state;
let modules;
let stubs = [
    'v-container',
    'v-app-bar'
]
let wrapper;
let checkoutStates;

beforeEach(() => {
    checkoutStates = {
        authentication: false,
        deliveryAddress: false,
        bankaccount: false,
        shipping: false,
        payment: false,
        summary: false
    }
    state = {}
    modules = {
        checkout: {
            namespaced: true,
            state() { return { checkoutStates } }
        }
    }
    store = new Vuex.Store({
        state,
        modules
    })
    wrapper = shallowMount(CheckoutNavbar, { localVue, stubs, store });
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })

    test('checkoutStates', () => {
        let navbuttons = wrapper.findAllComponents({ name: 'NavButton' });
        expect(navbuttons.wrappers).toHaveLength(6);
    })

    test('checkoutStates', () => {
        let navbuttons = wrapper.findAllComponents({ name: 'NavButton' });
        expect(navbuttons.wrappers[0].vm.name).toBe('Cancel');
        expect(navbuttons.wrappers[1].vm.name).toBe('Delivery address');
        expect(navbuttons.wrappers[2].vm.name).toBe('Bank Account');
        expect(navbuttons.wrappers[3].vm.name).toBe('Shipping');
        expect(navbuttons.wrappers[4].vm.name).toBe('Payment');
        expect(navbuttons.wrappers[5].vm.name).toBe('Summary');
    })
});