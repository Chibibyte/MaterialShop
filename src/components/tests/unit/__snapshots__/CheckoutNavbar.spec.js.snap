// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<v-app-bar-stub
  app=""
  class="accent"
>
  <v-container-stub
    app=""
    class="d-flex justify-space-around"
  >
    <navbutton-stub
      icon="mdi-window-close"
      name="Cancel"
      to="/"
    />
     
    <navbutton-stub
      disabled="true"
      icon="mdi-login-variant"
      name="Delivery address"
      to="/checkout/address"
    />
     
    <navbutton-stub
      disabled="true"
      icon="mdi-package-variant"
      name="Bank Account"
      to="/checkout/bankaccount"
    />
     
    <navbutton-stub
      disabled="true"
      icon="mdi-bank"
      name="Shipping"
      to="/checkout/shipping"
    />
     
    <navbutton-stub
      disabled="true"
      icon="mdi-truck"
      name="Payment"
      to="/checkout/payment"
    />
     
    <navbutton-stub
      disabled="true"
      icon="mdi-cash"
      name="Summary"
      to="/checkout/summary"
    />
  </v-container-stub>
</v-app-bar-stub>
`;
